econml/iv/dr/_dr.py

```diff
@@ -73,83 +246,1246 @@ def _effect_estimate(self, nuisances):
+
+        For integer/None inputs, if the treatment is discrete
+        :class:`~sklearn.model_selection.StratifiedKFold` is used, else,
+        :class:`~sklearn.model_selection.KFold` is used
+        (with a random shuffle in either case).
+
+        Unless an iterable is used, we call `split(concat[W, X], T)` to generate the splits. If all
+        W, X are None, then we call `split(ones((T.shape[0], 1)), T)`.
+
+    mc_iters: int, optional (default=None)
+        The number of times to rerun the first stage models to reduce the variance of the nuisances.
+
+    mc_agg: {'mean', 'median'}, optional (default='mean')
+        How to aggregate the nuisance value for each sample across the `mc_iters` monte carlo iterations of
+        cross-fitting.
+
+    random_state: int, :class:`~numpy.random.mtrand.RandomState` instance or None, optional (default=None)
+        If int, random_state is the seed used by the random number generator;
+        If :class:`~numpy.random.mtrand.RandomState` instance, random_state is the random number generator;
+        If None, the random number generator is the :class:`~numpy.random.mtrand.RandomState` instance used
+        by :mod:`np.random<numpy.random>`.
+    """
+
+    def __init__(self, *,
+                 model_y_xw="auto",
+                 model_t_xw="auto",
+                 model_z_xw="auto",
+                 model_t_xwz="auto",
+                 model_tz_xw="auto",
+                 prel_model_effect,
+                 projection=False,
+                 featurizer=None,
+                 fit_cate_intercept=True,
+                 cov_clip=0.1,
+                 opt_reweighted=False,
+                 discrete_instrument=False,
+                 discrete_treatment=False,
+                 categories='auto',
+                 cv=2,
+                 mc_iters=None,
+                 mc_agg='mean',
+                 random_state=None):
+        super().__init__(model_y_xw=model_y_xw,
+                         model_t_xw=model_t_xw,
+                         model_z_xw=model_z_xw,
+                         model_t_xwz=model_t_xwz,
+                         model_tz_xw=model_tz_xw,
+                         prel_model_effect=prel_model_effect,
+                         model_final=None,
+                         projection=projection,
+                         featurizer=featurizer,
+                         fit_cate_intercept=fit_cate_intercept,
+                         cov_clip=cov_clip,
+                         opt_reweighted=opt_reweighted,
+                         discrete_instrument=discrete_instrument,
+                         discrete_treatment=discrete_treatment,
+                         categories=categories,
+                         cv=cv,
+                         mc_iters=mc_iters,
+                         mc_agg=mc_agg,
+                         random_state=random_state)
+
+    def _gen_model_final(self):
+        return StatsModelsLinearRegression(fit_intercept=False)
```

Comment:
If we switch then this should be:
return StatsModelsLinearRegression(fit_intercept=self.fit_cate_intercept)